parts/k8s/kuberneteswindowssetup.ps1

```diff
@@ -139,17 +139,27 @@ try
     if ($true) {
         Write-Log "Provisioning $global:DockerServiceName... with IP $MasterIP"
 
-        Write-Log "apply telemetry data setting"
+        Write-Log "Apply telemetry data setting"
         Set-TelemetrySetting -WindowsTelemetryGUID $global:WindowsTelemetryGUID
 
-        Write-Log "resize os drive if possible"
+        Write-Log "Resize os drive if possible"
         Resize-OSDrive
 
-        Write-Log "install docker"
+        Write-Log "Install docker"
         Install-Docker -DockerVersion $global:DockerVersion
 
-        Write-Log "download kubelet binaries and unzip"
-        Get-KubeBinaries -KubeBinariesSASURL $global:KubeBinariesSASURL
+        Write-Log "Download kubelet binaries and unzip"
+        Get-KubePackage -KubeBinariesSASURL $global:KubeBinariesPackageSASURL
+
+        # this overwrite the binaries that are download from the custom packge with binaries 
+        # The custom package has a few files that are nessary for future steps (nssm.exe)
+        # this is a temporary work around to get the binaries until we depreciate 
+        # custom package and nssm.exe as defined in #3851.
+        if ($global:WindowsKubeBinariesURL){

```

Comment:
So before we were always calling `Get-KubeBinaries` and passing in `KubeBinariesSASURL`. Now we're only calling `Get-KubeBinaries` if `WindowsKubeBinariesURL` is non-empty. Is that the correct interpretation?

Is that change in implementation intentional?

Reply:
I renamed the old `Get-KubeBinaries` to `Get-KubePackage` to be more explicit in its functionality and created the new function with the old name.  I did this to make it more descriptive in what each function does.  `Get-KubePackage` will be removed in future commits as described in  #3851.

Reply:
This approach looks good to me